components/datadog/datadog.app.mjs

```diff
@@ -0,0 +1,213 @@
+      let page = 0;
+      let pageCount;
+      let perPage;
+      do {
+        const params = {
+          page,
+          query,
+        };
+        const {
+          monitors,
+          metadata,
+        } = await this._monitorsApi().searchMonitors(params);
+        for (const monitor of monitors) {
+          yield monitor;
+        }
+        ++page;
+        pageCount = metadata.pageCount;
+        perPage = metadata.perPage;
+      } while (pageCount === perPage);
+    },
+    async listMonitors(page, pageSize) {
+      return this._monitorsApi().listMonitors({
+        page,
+        pageSize,
+      });
+    },
+    async createWebhook(
+      url,
+      payloadFormat = null,
+      secretKey = uuid(),
+    ) {
+      const name = `pd-${uuid()}`;
+      const customHeaders = {
+        [this._webhookSecretKeyHeader()]: secretKey,
+      };
+      await this._webhooksApi().createWebhooksIntegration({
+        body: {
+          customHeaders: JSON.stringify(customHeaders),
+          payload: JSON.stringify(payloadFormat),
+          name,
+          url,
+        },
+      });
+      return {
+        name,
+        secretKey,
+      };
+    },
+    async deleteWebhook(webhookName) {
+      await this._webhooksApi().deleteWebhooksIntegration({
+        webhookName,
+      });
+    },
+    async addWebhookNotification(webhookName, monitorId) {
+      const { message } = await this._getMonitor(monitorId);
+      const webhookTagPattern = this._webhookTagPattern(webhookName);
+      if (new RegExp(webhookTagPattern).test(message)) {
+        // Monitor is already notifying this webhook
+        return;
+      }
+
+      const newMessage = `${message}\n${webhookTagPattern}`;
+      const monitorChanges = {
+        message: newMessage,
+      };
+      await this._editMonitor(monitorId, monitorChanges);
+    },
+    async removeWebhookNotifications(webhookName) {
+      // Users could have manually added this webhook in other monitors, or
+      // removed the webhook from the monitors specified as user props. Hence,
+      // we need to search through all the monitors that notify this webhook and
+      // remove the notification.
+      const webhookTagPattern = new RegExp(
+        `\n?${this._webhookTagPattern(webhookName)}`,
+      );
+      const monitorSearchResults = this._searchMonitors(webhookName);
+      for await (const monitorInfo of monitorSearchResults) {
+        const { id: monitorId } = monitorInfo;
+        const { message } = await this._getMonitor(monitorId);
+
+        if (!new RegExp(webhookTagPattern).test(message)) {
+          // Monitor is not notifying this webhook, skip it...
+          return;
+        }
+
+        const newMessage = message.replace(webhookTagPattern, "");
+        const monitorChanges = {
+          message: newMessage,
+        };
+        await this._editMonitor(monitorId, monitorChanges);
+      }
+    },
+    async listHosts() {
+      return this._hostsApi().listHosts();
```

Comment:
Should this method have a `params` parameter (used [here](https://github.com/PipedreamHQ/pipedream/blob/010ee88c5c6aa1031911b769fa1432a33f898073/components/datadog/datadog.app.mjs#L17))?
```suggestion
    async listHosts(params) {
      return this._hostsApi().listHosts(params);
```

Reply:
Addressed